service/kvmconfig/v6/key/key.go

```diff
@@ -124,18 +128,54 @@ func MasterHostPathVolumeDir(clusterID string, vmNumber string) string {
 }
 
 // MemoryQuantity returns a resource.Quantity that represents the memory to be used by the nodes.
-// It adds the memory from the node definition parameter to the additional memory defined by
-// defaultWorkerMemory.
-func MemoryQuantity(n v1alpha1.KVMConfigSpecKVMNode) (resource.Quantity, error) {
-	q, err := resource.ParseQuantity(n.Memory)
-	if err != nil {
-		return resource.Quantity{}, microerror.Maskf(err, "creating Memory quantity from node definition")
-	}
-	additionalMemory, err := resource.ParseQuantity(defaultWorkerMemory)
-	if err != nil {
-		return resource.Quantity{}, microerror.Maskf(err, "creating Memory quantity from addtional memory")
+// It adds the memory from the node definition parameter to the additional memory calculated on the node role
+func MemoryQuantity(n v1alpha1.KVMConfigSpecKVMNode, role string) (resource.Quantity, error) {
+	var q resource.Quantity
+	var err error
+	if role == MasterID {
+		q, err = resource.ParseQuantity(n.Memory)
+		if err != nil {
+			return resource.Quantity{}, microerror.Maskf(err, "creating Memory quantity from node definition")
+		}
+		additionalMemory, err := resource.ParseQuantity(baseMasterMemoryOverhead)
+		if err != nil {
+			return resource.Quantity{}, microerror.Maskf(err, "creating Memory quantity from addtional memory")
+		}
+		q.Add(additionalMemory)
+
+	} else if role == WorkerID {
+		i, err := strconv.Atoi(n.Memory[:len(n.Memory)-1])
+		if err != nil {
+			return resource.Quantity{}, microerror.Maskf(err, "calculating memory overhead multiplier")
+		}
+		// base worker memory calculated in MB
+		baseMemoryMB := strconv.Itoa(i * 1024)
+
+		q, err = resource.ParseQuantity(baseMemoryMB + "M")
```

Comment:
This could be just `q, err = resource.ParseQuantity(fmt.Sprintf("%dM", i * 1024))`. Or maybe even `q, err = resource.ParseQuantity(fmt.Sprintf("%dGi", i))`